components/brave_vpn/brave_vpn_service.cc

```diff
@@ -870,10 +870,37 @@ void BraveVpnService::OnSuspend() {
   // Set reconnection state in case if computer/laptop is going to sleep.
   // The disconnection event will be fired after waking up and we want to
   // restore the connection.
-  needs_connect_ = is_connected();
+  if (is_connected()) {
+    Disconnect();
+    reconnect_on_resume_ = true;
+  }
+  VLOG(2) << __func__
+          << " Should reconnect when resume:" << reconnect_on_resume_;
+}
+
+void BraveVpnService::OnResume() {
+#if BUILDFLAG(IS_MAC)
+  OnDNSChanged();
+#endif  // BUILDFLAG(IS_MAC)
+}
+
+void BraveVpnService::OnDNSChanged() {
+  if (!IsNetworkAvailable() ||
+      // This event is triggered before going to sleep while vpn is still
+      // active. Vpn is presented as CONNECTION_UNKNOWN and so we have to skip
+      // this to be notified only when default network active without VPN to
+      // reconnect.
+      net::NetworkChangeNotifier::GetConnectionType() ==
+          net::NetworkChangeNotifier::CONNECTION_UNKNOWN)
+    return;
+
+  VLOG(2) << __func__ << " Should reconnect:" << reconnect_on_resume_;
+  if (reconnect_on_resume_) {
+    Connect();
+    reconnect_on_resume_ = false;
```

Comment:
Instead of clearing `reconnect_on_resme_` here, how about doing it in `Connect()`?

Reply:
`Connect` can be called from another places and drop this flag, this actually happens with `need_reconnect_` in old solution

Reply:
Ok thanks!